lib/Doctrine/ORM/Mapping/MappingException.php

```diff
@@ -251,14 +251,13 @@ public static function nameIsMandatoryForSqlResultSetMapping($className)
         return new self("Result set mapping name on entity class '$className' is not defined.");
     }
 
-    /**
-     * @param string $fieldName
-     *
-     * @return MappingException
-     */
-    public static function oneToManyRequiresMappedBy($fieldName)
+    public static function oneToManyRequiresMappedBy(string $entityName, string $fieldName): MappingException
```

Comment:
```suggestion
    public static function oneToManyRequiresMappedBy(string $entityName, string $fieldName): self
```

This is a BC-break. If you think that class is not part of the BC promise, then maybe let's explicitly mark it as `@internal`?

Reply:
Yes its a BC break, but this exception is only used in `ClassMetadataInfo` in a code block that cannot be overwritten. Users are not expected to use it in any extension points themselves, such as a custom `MappingDriver`. `MappingException` cannot be `@internal` and adding that to every method seems a bit overhead or? I think there is no risk to ignore this.